plugins/forms/plugin.js

```diff
@@ -277,3 +277,16 @@ CKEDITOR.plugins.add( 'forms', {
 		}
 	}
 } );
+
+CKEDITOR.plugins.forms = {
+	/*
+	* Private helper function for setting dialog required value.
+	*
+	* @private
+	* @param {CKEDITOR.dom.element} element Element which required attribute is checked.
+	* */
+	_setupFormDialogRequired: function( element ) {
```

Comment:
This is incorrect implementation. According to the [HTML LS standard](https://html.spec.whatwg.org/multipage/input.html#the-required-attribute):

> The required attribute is a boolean attribute. When specified, the element is required.

[And further](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#boolean-attribute):

> A number of attributes are boolean attributes. The presence of a boolean attribute on an element represents the true value, and the absence of the attribute represents the false value.

Therefore the correct test is to check if the element has appropriate attribute (`element.hasAttribute( 'required' )`), as even incorrect values are treated as true.